src/wallet/wallets.go

```diff
@@ -139,11 +138,23 @@ func (wlts *Wallets) Remove(id string) {
 // Get returns wallet by wallet id
 func (wlts *Wallets) Get(wltID string) (Wallet, bool) {
 	if w, ok := (*wlts)[wltID]; ok {
-		return *w, true
+		return w.Copy(), true
 	}
 	return Wallet{}, false
 }
 
+// Update updates the given wallet, return error if not exist
+func (wlts *Wallets) Update(wltID string, updateFunc func(Wallet) Wallet) error {
+	w, ok := (*wlts)[wltID]
+	if !ok {
+		return errWalletNotExist(wltID)
+	}
+
+	newWlt := updateFunc(*w)
+	(*wlts)[wltID] = &newWlt
```

Comment:
is the pointer dereference necessary?

Reply:
Oh, yes, no need to use pointer (wlts *Wallets), the wallets is a map